doc/sphinx/proof-engine/ltac.rst

```diff
@@ -1083,45 +1567,49 @@ Testing boolean expressions
    .. exn:: Condition not satisfied.
       :undocumented:
 
-Proving a subgoal as a separate lemma
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Proving a subgoal as a separate lemma: abstract
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. tacn:: abstract @ltac_expr
-   :name: abstract
+.. tacn:: abstract @ltac_expr2 {? using @ident }
 
-   From the outside, :n:`abstract @ltac_expr` is the same as :n:`solve @ltac_expr`.
-   Internally it saves an auxiliary lemma called ``ident_subproofn`` where
-   ``ident`` is the name of the current goal and ``n`` is chosen so that this is
-   a fresh name. Such an auxiliary lemma is inlined in the final proof term.
+   Does a :tacn:`solve` :n:`[ @ltac_expr2 ]` and saves the subproof as an auxiliary lemma.
+   if :token:`ident` is specified, the lemma is saved with that name; otherwise
+   the lemma is saved with the name `NAME_subproofN` where
+   `NAME` is the name of the current goal and `N` is
+   a fresh name. The auxiliary lemma is inlined in the final proof term.
 
-   This tactical is useful with tactics such as :tacn:`omega` or
-   :tacn:`discriminate` that generate huge proof terms. With that tool the user
-   can avoid the explosion at time of the Save command without having to cut
-   manually the proof in smaller lemmas.
+   This is useful with tactics such as :tacn:`omega` or
+   :tacn:`discriminate` that generate huge proof terms with many intermediate
+   goals.  It can significantly reduce peak memory use.  In most cases it doesn't
+   have a significant impact on run time.  One case in which it can reduce run time
+   is when a tactic `foo` is known to always pass type checking when it
+   succeeds, such as in reflective proofs.  In this case, the idiom
+   ":tacn:`abstract` :tacn:`exact_no_check` `foo`" will save half the type
+   checking type time compared to ":tacn:`exact` `foo`".
 
    It may be useful to generate lemmas minimal w.r.t. the assumptions they
    depend on. This can be obtained thanks to the option below.
 
-   .. warning::
+   :tacn:`abstract` is an :token:`l3_tactic`.
 
-      The abstract tactic, while very useful, still has some known
-      limitations, see https://github.com/coq/coq/issues/9146 for more
-      details. Thus we recommend using it caution in some
-      "non-standard" contexts. In particular, ``abstract`` won't
-      properly work when used inside quotations ``ltac:(...)``, or
-      if used as part of typeclass resolution, it may produce wrong
-      terms when in universe polymorphic mode.
+   .. todo PR above: what "option below"??
```

Comment:
Same question. @JasonGross: do you know?

Reply:
bfbb9f063434623d7c3dac8aa4aaf64c4ec84373 added the text
```tex
It may be useful to generate lemmas minimal w.r.t. the assumptions they depend
on. This can be obtained thanks to the option below.

\begin{quote}
\optindex{Shrink Abstract}
{\tt Set Shrink Abstract}
\end{quote}

When set, all lemmas generated through \texttt{abstract {\tacexpr}} are
quantified only over the variables that appear in the term constructed by
\texttt{\tacexpr}.
```
The option text was removed with the removal of `Shrink Abstract` in #6903, but I guess the initial sentence was left behind.  It should be removed.  I believe the shrinkage now always happens, as can be confirmed with
```coq
Lemma foo : True -> True.
Proof.
  intros.
  abstract exact I.
Defined.
Print foo.
Print foo_subproof.
```